.github/workflow/ci.yml

```
@@ -0,0 +1,32 @@
+name: james potter
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    container:
+      image: idaholab/moose
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Build
+        run: |
+          mkdir build
+          cd build
+          cmake ..
+          make
+          echo "Running tests inside the container"
+
+      - name: Run Tests
+        run: |
+          cd ../
+          chmod +x run_test_cases.sh
+          ./run_test_cases.sh
```

A new `run` block will put you back in the initial directory, you shouldn't need to change directories.

You can also call this script without setting it be executable.  If you did want it to be executable, you make it that way and commit it as executable.
```suggestion
          source ./run_test_cases.sh
```